.github/workflows/reusable-build.yml

```diff
@@ -188,7 +188,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [14, 16]
+        node: [16]
```

Comment:
Why remove 14?

Reply:
> Why remove 14?

About this part its for 2 reasons:

It is recommend using node 16 in the contribution https://github.com/web-infra-dev/rspack/blob/main/CONTRIBUTING.md#install-nodejs  doc so I though it would make sense to align the version in the CI as well
pnpm version 8 requires node version 16 ++

Reply:
> pnpm version 8 requires node version 16 ++

I miss this. Rspack is guaranteed to be running in node14 and node16. If that's the case, I'm not sure we should update to pnpm@8.

Reply:
Sure no worries. I was just following the docs, let me know when you have updates 
Thanks

Reply:
> > pnpm version 8 requires node version 16 ++
> 
> I miss this. Rspack is guaranteed to be running in node14 and node16. If that's the case, I'm not sure we should update to pnpm@8.

Seems that the change brings some complications
 I can close it with no concern if its the case

Reply:
We would have a meeting tomorrow to discuss this PR and related topics. You could keep the PR open until the team come up with some conclusions.

Reply:
@hyf0 sure here it is https://github.com/web-infra-dev/rspack/pull/2780